Lib/test/test_inspect/test_inspect.py

```diff
@@ -2302,6 +2304,13 @@ def __dict__(self):
         self.assertEqual(inspect.getattr_static(foo, 'a'), 3)
         self.assertFalse(test.called)
 
+        class Bar(Foo): pass
+
+        bar = Bar()
+        bar.a = 5
+        self.assertEqual(inspect.getattr_static(bar, 'a'), 3)
+        self.assertFalse(test.called)
+
```

Comment:
Did this case get broken in the development of this PR, or what motivated adding this?

Reply:
Nah, I just got scared at some point while working on the PR that the cache would lead to incorrect behaviour for instances of subclasses of a class that had a shadowed dict. I think my reasoning at that point was bogus, but I wrote the test to be sure, and I think an extra test here probably doesn't hurt given the subtlety of this code!